features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/net/BatchesToSegmentsMapperTest.kt

```diff
@@ -48,158 +48,150 @@ internal class BatchesToSegmentsMapperTest {
     }
 
     @Test
-    fun `M generate segment to jsonSegment pair W map`(forge: Forge) {
+    fun `M group the records into segments by context W map`(forge: Forge) {
         // Given
-        val rumContext: SessionReplayRumContext = forge.getForgery()
         val fakeRecordsSize = forge.anInt(min = 10, max = 20)
-        val fakeRecords: List<MobileSegment.MobileRecord> = forge.aList(fakeRecordsSize) {
+        val fakeRecords: List<MobileSegment.MobileRecord> = forge.aList<MobileSegment.MobileRecord>(fakeRecordsSize) {
```

Comment:
the type is already specified on the left, why it is needed here as well? same for similar lines below.

Reply:
actually it requires the type on the right side also as it doesn't know how to resolve it in the lambda. It happened to me before.